src/schemas/m2cms.ts

```diff
@@ -0,0 +1,152 @@
+import process from 'process';
+import fs from 'node:fs';
+import path from 'node:path';
+import Typescript, {
+    type Node,
+    type InterfaceDeclaration,
+    type PropertySignature,
+    type Identifier
+} from 'typescript';
+import YAML from 'yaml';
+
+import Debug from 'debug';
+const DEBUG = Debug('m2cms')
+
+const KIND = (e: Node | undefined ) => e && Typescript.SyntaxKind[e.kind]
+
+interface Field {
+    label: string;
+    name: string;
+    widget: string;
+    fields?: Field[];
+}
+
+interface Collection {
+    name: string;
+    label: string;
+    folder: string;
+    widget: string;
+    create: boolean;
+    slug: string;
+    fields: Field[];
+}
+
+interface Config {
+    local_backend?: boolean;
+    backend?: {
+        name: string;
+        repo: string;
+        branch?: string;
+    }
+    publish_mode?: string;
+    media_folder: string;
+    collections: Collection[];
+}
+
+
+function parsePropertySignature(ps: PropertySignature, ancestors: string[] = []) : Field | null {
+    const name = (ps.name as Identifier).escapedText.toString();
```

Comment:
> There is also a cast which I don't quite get in that function, and might be worth reviewing.

https://typestrong.org/typedoc-auto-docs/typedoc/interfaces/TypeScript.PropertySignature.html#name
https://typestrong.org/typedoc-auto-docs/typedoc/modules/TypeScript.html#PropertyName

ps.name is of type PropertyName that can be a lot of thing, here we really need it to be an Identifier. I'm not sure what's the correct way to tell that.

Reply:
Isn't there an `isIdentifier` type guard?

Reply:
i can't find it in the docs